src/test/groovy/unit/org/pih/warehouse/api/ProductApiControllerSpec.groovy

```diff
@@ -0,0 +1,121 @@
+package unit.org.pih.warehouse.api
+
+import grails.orm.PagedResultList
+import grails.testing.gorm.DataTest
+import grails.testing.web.controllers.ControllerUnitTest
+import java.util.stream.IntStream
+import org.grails.plugins.testing.GrailsMockHttpServletResponse
+import org.grails.web.json.JSONElement
+import org.grails.web.json.JSONObject
+import org.hibernate.Criteria
+import org.json.JSONArray
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Unroll
+
+import org.pih.warehouse.api.ProductApiController
+import org.pih.warehouse.core.Location
+import org.pih.warehouse.forecasting.ForecastingService
+import org.pih.warehouse.product.Product
+import org.pih.warehouse.product.ProductService
+
+@Unroll
+class ProductApiControllerSpec extends Specification implements DataTest, ControllerUnitTest<ProductApiController> {
+
+	@Shared private def forecastingServiceStub
+	@Shared private def productServiceStub
+
+    void setupSpec() {
+		mockDomains(Product, Location)
+    }
+
+	void setup() {
```

Comment:
there seems to be too much indentation (between setupSpec and the setup methods)

Reply:
yes I think this is because I mixed tabs and spaces. From what I can see our code standard is 4 spaces so I'll reform the file. I used tabs for years so it's going to take a minute to shake that convention out of me hehe. 

(As a side note, we should really have a standardized prefs file that we can import to Intellij so that all our setups stay in sync.)